packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -195,6 +196,25 @@ const DataGrid = forwardRef((props, ref) => {
   } = props;
 
   const displayedColumns = (hasSelectableRows ? [WorklistDataGridUtils.ROW_SELECTION_COLUMN] : []).concat(pinnedColumns).concat(overflowColumns);
+  // Create new displayedColumns object to account for column spans
+  const displayedColumnsWithColumnSpan = useMemo(() => {
+    let i = 0;
+    const newDisplayedColumns = [];
+    displayedColumns.forEach((column) => {
```

Comment:
Would a reducer be a better pattern for this functionality?  It seems like the accumulator could help simplify this logic similar to how we deal with the row indexes for sections in the table.